### DISCUSSION TITLE
The PRACH detector will not meet the performance requirements with the configuration {Format 0, ZCZ 0, SCS 1.25kHz, Rx ports 1}.

### DISCUSSION THREAD

**Discussion by Zplusless** [srsran/srsRAN_Project#414] [General Help] [answered]
![image](https://github.com/srsran/srsRAN_Project/assets/10062702/12f80ef1-ff55-4616-8733-242c962e3b41)

**"The PRACH detector will not meet the performance requirements with the configuration {Format 0, ZCZ 0, SCS 1.25kHz, Rx ports 1}."**

I have tested five recent  master versions of gnb, the above information appears since version 55c984b commited on 20231207.

|version               |has this info            |commit date|
|  ----  | ----  | ----  |
|5e6f50a              |    no||
|374200d             |    no||
|55c984b             |    yes                       | 20231207|
|dcd905c             |    yes                       | 20231222|
|0b2702c              |   yes                       | 20240115|

sys info:
![image](https://github.com/srsran/srsRAN_Project/assets/10062702/23b07e40-0eb7-4636-acc0-efae620a4568)

[gnb_zmq-fusion.yaml.txt](https://github.com/srsran/srsRAN_Project/files/13987896/gnb_zmq-fusion.yaml.txt)

**Comment by dvdgrgrtt** [comment, marked as answer]
Hi @Zplusless,
thanks for pointing this out. However, this is not a regression - in old versions of the software, we only had a rough PRACH detector. Around the end of November, we improved the detector, fine-tuned the detection thresholds and introduced these warnings.

The problem is that the PRACH has hundreds of different configurations and it is impossible for us to ensure "perfect" performance for all of them. Specifically, the NR standard requires a 99% probability of detection and a 0.1% probability of false alarm. The warning you get is only saying that we don't reach such high requirements for the requested configuration, but, most of the times, you won't notice the difference - for instance, with the configuration you tried, the estimated performance is 98% probability of detection. Finally, for your info, there are also a number of configuration whose performance is too low for the gNB to work properly (for instance Format 0, ZCZ 14 or 15, 1 Rx port), but in that case the warning will be replaced by an error that causes the gNB application to stop.

Hope this helps!

**Reply by Zplusless** [reply]
Hi @dvdgrgrtt, Thanks for your explanation! I will retain the existing configuration and ingore the warning as long as the gNB is functioning properly. Thanks again!
